src/Illuminate/Database/Eloquent/SoftDeletes.php

```diff
@@ -33,7 +33,9 @@ public static function bootSoftDeletes()
      */
     public function initializeSoftDeletes()
     {
-        $this->dates[] = $this->getDeletedAtColumn();
+        if (! isset($this->casts[$this->getDeletedAtColumn()])) {
```

Comment:
Hmmm, this change is unfortunate. I wonder if there's a better way to implement traits wanting to define casts, in general?

Reply:
What is _unfortunate_ exactly? The direct access to `casts`?

Reply:
We could have an abstract method within them that exposes this data, but I do not find a reason for it since these traits usually manipulate states.  I will be also curious to see what's the taken path for this.

Reply:
One can use:
```php
$this->casts = array_merge($casts, $this->casts);
```

But since this can be a more common use case maybe we could update the [mergeCasts](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L492) method:
```php
    /**
     * Merge new casts with existing casts on the model.
     *
     * @param  array  $casts
     * @param  bool  $asDefault
     * @return void
     */
    public function mergeCasts($casts, bool $asDefault = false)
    {
        $this->casts = $asDefault ? array_merge($casts, $this->casts) : array_merge($this->casts, $casts);
    }
```

Alternatively we could introduce a mergeDefaultCasts method but in my opinion we should avoid introducing new methods in the model if it can be avoided.